x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportInferenceUsageAction.java

```diff
@@ -62,17 +81,207 @@ protected void localClusterStateOperation(
+     */
+    private void addStatsForDefaultModels(
+        Map<ServiceAndTaskType, Map<String, List<InferenceFieldMetadata>>> inferenceFieldsByIndexServiceAndTask,
+        List<ModelConfigurations> endpoints,
+        Map<String, ModelStats> endpointStats
+    ) {
+        Map<String, String> endpointIdToModelId = endpoints.stream()
+            .filter(endpoint -> endpoint.getServiceSettings().modelId() != null)
+            .collect(Collectors.toMap(ModelConfigurations::getInferenceEntityId, e -> stripLinuxSuffix(e.getServiceSettings().modelId())));
+        Map<DefaultModelStatsKey, Long> defaultModelsToEndpointCount = createDefaultStatsKeysWithEndpointCounts(endpoints);
+        for (Map.Entry<DefaultModelStatsKey, Long> defaultModelStatsKeyToEndpointCount : defaultModelsToEndpointCount.entrySet()) {
+            DefaultModelStatsKey statKey = defaultModelStatsKeyToEndpointCount.getKey();
+            Map<String, List<InferenceFieldMetadata>> fieldsByIndex = inferenceFieldsByIndexServiceAndTask.getOrDefault(
+                new ServiceAndTaskType(statKey.service, statKey.taskType),
+                Map.of()
+            );
+            // Now that we have all inference fields for this service and task type, we want to keep only the ones that
+            // reference the current default model.
+            fieldsByIndex = filterFields(fieldsByIndex, f -> statKey.modelId.equals(endpointIdToModelId.get(f.getInferenceId())));
+            ModelStats stats = new ModelStats(statKey.toString(), statKey.taskType, defaultModelStatsKeyToEndpointCount.getValue());
+            addSemanticTextStats(fieldsByIndex, stats);
+            endpointStats.put(statKey.toString(), stats);
+        }
+    }
+
+    private Map<DefaultModelStatsKey, Long> createDefaultStatsKeysWithEndpointCounts(List<ModelConfigurations> endpoints) {
+        // We consider models to be default if they are associated with a default inference endpoint.
+        // Note that endpoints could have a null model id, in which case we don't consider them default as this
+        // may only happen for external services.
+        Set<String> modelIds = endpoints.stream()
+            .filter(endpoint -> modelRegistry.containsDefaultConfigId(endpoint.getInferenceEntityId()))
+            .filter(endpoint -> endpoint.getServiceSettings().modelId() != null)
+            .map(endpoint -> stripLinuxSuffix(endpoint.getServiceSettings().modelId()))
+            .collect(Collectors.toSet());
+        return endpoints.stream()
+            .filter(endpoint -> endpoint.getServiceSettings().modelId() != null)
+            .filter(endpoint -> modelIds.contains(stripLinuxSuffix(endpoint.getServiceSettings().modelId())))
+            .map(
+                endpoint -> new DefaultModelStatsKey(
+                    endpoint.getService(),
+                    endpoint.getTaskType(),
+                    stripLinuxSuffix(endpoint.getServiceSettings().modelId())
+                )
+            )
```

Comment:
Can you delete the stream to build the modelIds set above and include the `modelRegistry.containsDefaultConfigId(...` filter here. 

Basically, is this equivalent? 

```suggestion
        return endpoints.stream()
            .filter(endpoint -> endpoint.getServiceSettings().modelId() != null)
            .filter(endpoint -> modelRegistry.containsDefaultConfigId(endpoint.getInferenceEntityId()))
            .map(
                endpoint -> new DefaultModelStatsKey(
                    endpoint.getService(),
                    endpoint.getTaskType(),
                    stripLinuxSuffix(endpoint.getServiceSettings().modelId())
                )
            )
```